.github/workflows/build-and-release-to-spin.yml

```diff
@@ -17,6 +17,14 @@ env:
   IS_PROD_RELEASE: ${{ startsWith(github.ref, 'refs/tags/v') }}
 
 jobs:
+  setup-variables:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set variables
+        run: |
+          echo "IS_PROD_RELEASE=${{ startsWith(github.ref, 'refs/tags/v') }}" >> "$GITHUB_ENV"
+          echo "RANCHER_NAMESPACE=${{ startsWith(github.ref, 'refs/tags/v') && 'nmdc' || 'nmdc-dev' }}" >> "$GITHUB_ENV"
```

Comment:
Post-merge note to myself:

`startsWith` here is a GitHub Actions Expression: https://docs.github.com/en/actions/learn-github-actions/expressions#startswith

`&&` is an operator: https://docs.github.com/en/actions/learn-github-actions/expressions#operators

This "ternary operator"-like behavior is documented in the following example in the GitHub Actions Expressions documentation: https://docs.github.com/en/actions/learn-github-actions/expressions#example